assets/src/components/resource/editors/PurposeGroupEditor.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { PropsWithChildren } from 'react';
```

Comment:
Since we are starting to have more and more groups, I renamed this specific group to be PurposeGroupEditor to avoid confusion that this could be more general purpose that it actually is. The components and logic in here are specific to "purpose groups"